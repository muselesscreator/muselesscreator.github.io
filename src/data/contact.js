import { StrictDict } from '@muselesscreator/strict-dict';
import avatar from '../assets/ampersand.png';

const contactInfo = ({
  name: 'Ben Warzeski',
  email: 'bwarzeski@gmail.com',
  title: 'Full-Stack Software Engineer',
  phone: '+1 (941) 387-5092',
  phoneRaw: '19413875092',
  github: 'https://www.github.com/muselesscreator',
  linkedIn: 'https://www.linkedin.com/in/benjamin-warzeski-69111518/',
  avatar,
});

export default contactInfo;
