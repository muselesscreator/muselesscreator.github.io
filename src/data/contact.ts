import { StrictDict } from '@muselesscreator/strict-dict';
import avatar from '../assets/ampersand.png';

export type ContactInfo = {
  name: string;
  email: string;
  title: string;
  phone: string;
  phoneRaw: string;
  github: string;
  linkedIn: string;
  avatar: string;
};

const contactInfo = StrictDict({
  name: 'Ben Warzeski',
  email: 'bwarzesk@gmail.com',
  title: 'Full-Stack Software Engineer',
  phone: '+1 (941) 387-5092',
  phoneRaw: '19413875092',
  github: 'https://www.github.com/muselesscreator',
  linkedIn: 'https://www.linkedin.com/in/benjamin-warzeski-69111518/',
  avatar,
}) as ContactInfo;

export default contactInfo;
