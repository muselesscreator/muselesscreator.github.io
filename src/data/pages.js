import { StrictDict } from '@muselesscreator/strict-dict';

export const pages = StrictDict({
  about: 'about',
  contact: 'contact',
  resume: 'resume',
  portfolio: 'portfolio',
  blog: 'blog',
});

export const pageNames = StrictDict({
  [pages.about]: 'About',
  [pages.contact]: 'Contact',
  [pages.resume]: 'Resume',
  [pages.portfolio]: 'Portfolio',
  [pages.blog]: 'Blog',
});

export default pages;
