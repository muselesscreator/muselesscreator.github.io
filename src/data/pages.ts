import { StrictDict } from '@muselesscreator/strict-dict';

export const pages = StrictDict({
  about: 'about',
  contact: 'contact',
  resume: 'resume',
  portfolio: 'portfolio',
  demo: 'demo',
  blog: 'blog',
}) as Record<string, string>;

export const pageNames = StrictDict({
  [pages.about]: 'About',
  [pages.contact]: 'Contact',
  [pages.resume]: 'Resume',
  [pages.portfolio]: 'Portfolio',
  [pages.blog]: 'Blog',
  [pages.demo]: 'Demos',
}) as Record<string, string>;

export default pages;
