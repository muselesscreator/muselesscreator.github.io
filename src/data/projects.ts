import gradebookImg from '../assets/gradebook.png';
import learnerDashboardImg from '../assets/learnerDashboard.png';
import oraImg from '../assets/ora.png';
import strictDictImg from '../assets/strictDict.png';
import shallowSnapshotImg from '../assets/shallowSnapshot.png';
import useKeyedStateImg from '../assets/useKeyedState.png';
import oraGradingImg from '../assets/oraGrading.png';

import { StrictDict } from '@muselesscreator/strict-dict';
type ProjectData = {
  title: string;
  description: string;
  category: string;
  githubUrl: string;
  githubRepo: string;
  packageName?: string;
  packageUrl?: string;
  screenshots?: string[];
  details?: string;
  image?: string;
  projectUrl: string;
};

export const ProjectCategories = StrictDict({
  openSource: 'Open Source Contributions',
  public: 'Public Packages',
  all: 'All',
}) as Record<string, string>;

const projects: ProjectData[] = [
  {
    title: '@muselesscreator/react-shallow-snapshot',
    description: 'A shallow snapshot testing utility for React',
    category: ProjectCategories.public,
    githubUrl: 'https://www.github.com/muselesscreator/react-shallow-snapshot',
    githubRepo: 'muselesscreator/react-shallow-snapshot',
    packageName: '@muselesscreator/react-shallow-snapshot',
    packageUrl: 'https://www.npmjs.com/package/@muselesscreator/react-shallow-snapshot',
    image: shallowSnapshotImg,
    projectUrl: 'https://www.npmjs.com/package/@muselesscreator/react-shallow-snapshot',
  },
  {
    title: '@muselesscreator/strict-dict',
    description: 'A key-safe object wrapper for JS',
    category: ProjectCategories.public,
    githubUrl: 'https://www.github.com/muselesscreator/strict-dict',
    githubRepo: 'muselesscreator/strict-dict',
    packageName: '@muselesscreator/strict-dict',
    packageUrl: 'https://www.npmjs.com/package/@muselesscreator/strict-dict',
    details: 'A key-safe dictionary for JavaScript.  This package is used in all of my projects and is available for anyone to use.',
    image: strictDictImg,
    projectUrl: 'https://www.npmjs.com/package/@muselesscreator/strict-dict',
  },
  {
    title: '@muselesscreator/use-keyed-state',
    description: 'A more testable React hook for keyed state management',
    category: ProjectCategories.public,
    githubUrl: 'https://www.github.com/muselesscreator/use-keyed-state',
    githubRepo: 'muselesscreator/use-keyed-state',
    packageName: '@muselesscreator/use-keyed-state',
    packageUrl: 'https://www.npmjs.com/package/@muselesscreator/use-keyed-state',
    projectUrl: 'https://www.npmjs.com/package/@muselesscreator/use-keyed-state',
    image: useKeyedStateImg,
  },
  {
    title: 'EdX Gradebook App Re-Architecture',
    description: 'Grade management app for EdX Course Staff',
    category: ProjectCategories.openSource,
    githubUrl: 'https://www.github.com/openedx/frontend-app-gradebook',
    githubRepo: 'openedx/frontend-app-gradebook',
    packageName: '@edx/frontend-app-gradebook',
    packageUrl: 'npmjs.com/package/@edx/frontend-app-gradebook',
    details: 'I led a team effort to update and fully re-architect the old Gradebook app with modern React practices and added theme-ability and test coverage.',
    image: gradebookImg,
    projectUrl: 'https://www.github.com/openedx/frontend-app-gradebook',
  },
  {
    title: 'EdX Open Response Assessment Staff Grader',
    description: 'Staff-Grading tool for EdX short-answer and file-submission questions',
    category: ProjectCategories.openSource,
    githubUrl: 'https://www.github.com/openedx/frontend-app-ora-grading',
    githubRepo: 'openedx/frontend-app-ora-grading',
    packageName: '@edx/frontend-app-ora-grading',
    packageUrl: 'npmjs.com/package/@edx/frontend-app-ora-grading',
    details: 'Served as Technical Lead for the development of a new grading interface which drastically reduced the time, effort, and frustration involved in grading of one of the most-used problem types in the system.  Ensured a bug-free release of code with 100% test coverage.',
    image: oraGradingImg,
    projectUrl: 'https://www.github.com/openedx/frontend-app-ora-grading',
  },
  {
    title: 'EdX Learner Home Redesign',
    description: 'Landing page for learner course listings',
    category: ProjectCategories.openSource,
    githubUrl: 'https://www.github.com/openedx/frontend-app-learner-dashboard',
    githubRepo: 'openedx/frontend-app-learner-dashboard',
    packageName: '@edx/frontend-app-learner-dashboard',
    packageUrl: 'npmjs.com/package/@edx/frontend-app-learner-dashboard',
    details: `Served as Technical Lead and Project Manager for redesign of the Learner Home web interface (logged-in learner’s course listing page).  The new experience is faster to load, more configurable, more theme-able, and fully internationalized, while providing a much easier interface for adding revenue-significant upsell interfaces (suggested courses, upsell links, “where to go from here?”)`,
    image: learnerDashboardImg,
    projectUrl: 'https://www.github.com/openedx/frontend-app-learner-dashboard',
  },
  {
    title: 'EdX Open Response Assessment Learning Block Redesign',
    description: 'Learning block for all short-answer and file-submission questions',
    category: ProjectCategories.openSource,
    githubUrl: 'https://www.github.com/openedx/frontend-app-ora',
    githubRepo: 'openedx/frontend-app-ora',
    packageName: '@edx/frontend-app-ora',
    packageUrl: 'npmjs.com/package/@edx/frontend-app-ora',
    details: 'Served as Technical Lead and Project Manager for the full redesign of theOpen Response Assessment learning block, migrating legacy (backboneJS/jquery) experience to a modern experience using React and theme-able componentry.',
    image: oraImg,
    projectUrl: 'https://www.github.com/openedx/frontend-app-ora',
  },
];

export default projects;
