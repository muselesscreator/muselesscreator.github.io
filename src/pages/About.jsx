import 'react';

const About = () => {
  return (
    <article className="about active">
      <header>
        <h2 className="h2 article-title">About me</h2>
      </header>
      <div className="about-text">
        <p>With over a decade of software engineering, I specialize in helping teams produce solid, testable frontend-code and maintainable agile frontend development processes, as well as experience in building and managing communities devoted to frontend communication, coordination, and training.</p>
        <p>I have experience guiding large projects from conception, through ideation and refinement, implementation, validation, release, and maintenance, as well as coming into existing projects and codebases and updating them to a consistent level of quality.</p>
        <p>My background in SQA and the odd variety of engineering tech my career has exposed me to give me a mindset and focus on maintainability of code and practices above all else.  I like to spend a bit of thoughtful effort now to make less work for everyone later.</p>
        <p>My core goal in any code-base is not only to achieve excellent code, but specifically to make achieving excellence as easy as possible.  I have high standards for my code and the projects I own, but I also endeavor to provide tools, workflows, and utilities to allow developers to quickly build well-designed, easily-maintained, and highly/deeply tested code.</p>
        <p>It has also been my experience that Agile workflows and processes are very effective for managing the engineering process, and have been pleased to be able to lead and teach multiple teams in development and refinement of effective Agile practices, with a focus on iteration, clarity of ceremonies, and focus/efficacy of meetings.</p>
        <ul>
          <li>Senior Frontend/Full-Stack Software Engineer with 8+ years of experience with Python and Full-Stack Python/JavaScript Web development, with a focus on user interfaces, APIs, and testing methodologies.</li>
          <li>Experience developing in a variety of architectures, including web, mobile, asynchronous server applications, and  2D/3D gaming.</li>
          <li>Experience self-managing and leading on projects with complex requirements across departments and generating well defined, fully-featured, well-tested code.</li>
          <li>Experience with test automation, database design, and API design.</li>
        </ul>
      </div>
    </article>
  );
};

export default About;
