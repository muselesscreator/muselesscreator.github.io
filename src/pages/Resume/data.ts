import 'react';
import { StrictDict } from '@muselesscreator/strict-dict';

export type Field = ({
  type: 'text',
  text: string,
} | {
  type: 'list',
  title?: string,
  items: Field[],
});

export type ResumeSectionType = {
  items: {
    date: string,
    title: string,
    fields: Field[],
  }[],
};

type Resume = {
  resume: {
    education: ResumeSectionType,
    experience: ResumeSectionType,
  },
};


const pageData = StrictDict({
  resume: {
    education: {
      items: [
        {
          title: 'Tulane University',
          date: '2006 — 2011',
          fields: [
            { type: 'text', text: 'Bachelor of Science in Mathematics with minors in Music and "Music, Science and Technology".' },
          ],
        },
      ],
    },
    experience: {
      items: [
        {
          title: 'Senior Software Engineer II, UI-Lead - EdX/2U Inc, Boston, MA',
          date: '2020 - 2024',
          fields: [
            { type: 'text', text: 'I served as the Frontend technical leadership for my squad across formation and delivery of multiple projects, as well as serving as the Project Manager for later projects, including the Learner Home and ORA redesign projects.' },
            { type: 'text', text: 'My core priorities and contributions outside of project delivery came in the form of testing, community, and training.' },
            { type: 'list', title: 'Community-Building', items: [
              { type: 'text', text: 'Organized and managed the Fedx (frontend-at-edx) community, integrating rotating leadership, a manage-able charter, engineering leadership support/sponsorship, and well-defined responsibilities.' },
              { type: 'text', text: 'This community serves as the hub for frontend development across teams at a company with no formal frontend organization.  This provided us a place and avenue for discussing/resolving tricky “glue” type problems that affected many teams, but none specifically enough to get funding otherwise (dependency upgrades, pattern updates, adoption of new technology, definition and consolidation of coding standards).'},
            ] },
            { type: 'list', title: 'Training', items: [
              { type: 'text', text: 'In addition to training up a junior frontend dev on my team as well as training my backend devs to be comfortable working quickly and effectively in our React code, I spent a great deal of time training up developers on other teams on frontend development and testing techniques.' },
              { type: 'text', text: 'Organized and managed the frontend study group, and integrated it with the fedx community.  I wrote and ran a variety of training sessions on testing, hooks, react development, internal tools, typescript, and more.' },
            ] },
            { type: 'list', title: 'Testing', items: [
              { type: 'text', text: 'I maintain a high standard of testing in my projects, with a tendency to release with close to 100% test coverage.  I have a focus on accurate/easy unit tests that provide maximum value for minimum creation/maintenance time.' },
              { type: 'text', text: 'As part of my testing pattern, I have developed a testing library to serve as a replacement for Enzyme, which is getting deprecated in future versions of React, which has begun being used by members of the community to mass-overhaul testing patterns.' },
            ] },
          ],
        }, {
          title: 'Senior Software Engineer - ReadCoor Inc, Boston, MA',
          date: '2018 - 2020',
          fields: [
            { type: 'list', items: [
              { type: 'text', text: 'Facilitated technology comparison process across teams to determine in which new UI technology the disparate  User Interfaces at the company should work towards being re-written, managing to acquire broad buy-in  across multiple development teams with very different focuses and existing JS workflows.  • Succeeded in pushing to implement a stable Javascript development workflow, with documented coding  guidelines, UI-generation scripts, reusable plug-in JS components and modules, and a shared theming.' },
              { type: 'text', text: 'Lead re-write of multiple existing user interfaces into React/Redux.' },
              { type: 'text', text: 'Lead the requirements generation and implementation of the Customer Device Interface, providing control  over all the parts of the genetic sequencing scan setup and run procedure, with input from and co-ordination  with UX designer. As part of this process, I helped lead the generation of initial design and requirements,  provided 90% of JS implementation, facilitated/lead backend API updates/integration, and generated a full  set of requirements in Testlink along with a development manual test plan for engineer testers.' },
            ] },
          ],
        }, {
          title: 'Software Engineer, UI Lead - Optimus Ride, Boston, MA',
          date: '2017 — 2018',
          fields: [
            { type: 'list', items: [
              { type: 'text', text: 'As Lead UI Developer, implemented web-based customer-facing User Interfaces in React/Redux, targeted at  many levels of customer, including rider, dispatchers, vehicle operators, and fleet managers.' },
              { type: 'text', text: 'As Scrum Master for Fleet Software team: designed and enforced an agile system to bring structure to a  disorganized team. Implemented 2-week sprints, with regular agile meetings, resulting in improved output,  and vastly improved targeting of priorities by enforcing a structure around requirements-gathering and  problem definition from stakeholders.' },
              { type: 'text', text: 'As SW Engineer, built python backend services in support of web user interfaces, allowing tracking of a  meta-contextual state across system components and tracking/controlling the data driving the user  experience workflow throughout and across all user levels/types.' },
            ]}
          ],
        }, {
          title: 'Software Engineer, Software Engineer II - Bio-Rad, Boston MA',
          date: '2014 — 2017',
          fields: [
            { type: 'list', items: [
              { type: 'text', text: 'As Project Manager for the Customer User Interface: managed coordination between manufacturing,  marketing, biology, and software departments, along with an external UX contracting firm, to produce a  touch-screen user interface for the MonoRAyL system.' },
              { type: 'text', text: 'As Lead Developer for the Customer User Interface: produced a touch-screen web app based around a  Python web server, responsible for guiding the user through the steps of an experiment, while also managing  the hardware interaction, data access, user management, and software/firmware updates.' },
              { type: 'text', text: 'As Lead Developer for the Internal Development User Interface: produced a complex, robust one-page app  providing control over all aspects of the complex biomedical microfluidic device:' },
              { type: 'list', items: [
                { type: 'text', text: 'Provided controls and status readouts for over 40 physical sensors and interfaces, including: electrodes,  cameras, stepper motors, thermal plates, pressure pumps/valves, lasers, and LEDs' },
                { type: 'text', text: 'Provided sub-tools for manufacturing and software teams allowed management of all running sub processes and internal flags for the system as well as interactive tools for running end-of-line testing and  managing machine configurations.' },
                { type: 'text', text: 'Developed and maintained tools and APIs for inter-process communication, database management and interaction, unit testing, asynchronous process coordination, and hardware interactions.' },
              ]},
            ]},
          ],
        }, {
          title: 'Co-Founder/Lead Developer - Nerd Pile Gaming, Boston, MA',
          date: '2013 — 2016',
          fields: [
            { type: 'text', text: 'For the studio’s first iOS/Android release, Farmageddon' },
            { type: 'list', items: [
              { type: 'text', text: 'Lead Developer, implementing core gameplay APIs in Lua using CoronaSDK.' },
              { type: 'text', text: 'Lead Sound Engineer, producing sound effects, voice overs and seven music compositions.' },
            ] },
            { type: 'text', text: 'Post-Farmageddon' },
            { type: 'list', items: [
              { type: 'text', text: 'Lead Developer for further projects in Unity' },
              { type: 'text', text: 'Developed 2D and 3D interfaces for a variety of puzzle games' },
              { type: 'text', text: 'Lead Sound Engineer, four music compositions for PuzzlePlanets' },
            ] },
          ],
        },
      ],
    },
  },
});

export default pageData as Resume;
