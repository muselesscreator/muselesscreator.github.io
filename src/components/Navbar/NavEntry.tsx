import classNames from 'classnames';
import { NavLink } from 'react-router-dom';
import { pageNames } from '~/data/pages';


export type NavEntryProps = {
  page: string;
  isActive: boolean;
};

const NavEntry = ({ page, isActive }: NavEntryProps) => (
  <li key={page} className="navbar-item">
    <NavLink
      className={classNames('navbar-link', { active: isActive })}
      to={`/${page}`}
    >
    {pageNames[page]}
    </NavLink>
  </li>
);

export default NavEntry;
