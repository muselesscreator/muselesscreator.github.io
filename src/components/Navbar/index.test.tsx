import {
  vi,
  describe,
  test,
  expect,
} from 'vitest';

import { shallow } from '@muselesscreator/react-shallow-snapshot';

import { pages } from '~/data/pages';

import NavBar, { PAGE_ORDER } from '.';
import { NavEntryProps } from './NavEntry';

vi.mock('./NavEntry', () => ({ default: 'NavEntry' }));

type NavEntry = {
  props: NavEntryProps;
};

describe('NavBar component', () => {
  test('about page snapshot', () => {
    const wrapper = shallow(<NavBar  activePage={pages.about} />);
    expect(wrapper.snapshot).toMatchSnapshot();
  });
  describe('Nav entries (all should be included, and activePage should be active)', () => {
    PAGE_ORDER.forEach((page) => {
      test(page, () => {
        const wrapper = shallow(<NavBar  activePage={pages[page]} />);
        const navEntries = wrapper.instance.findByType('NavEntry');
        expect(navEntries).toHaveLength(PAGE_ORDER.length);
        navEntries.forEach((navEntry: NavEntry, index: number) => {
          expect(navEntry.props).toEqual({
            page: pages[PAGE_ORDER[index]],
            isActive: pages[PAGE_ORDER[index]] === pages[page],
          });
        });
      });
    });
  });
});
