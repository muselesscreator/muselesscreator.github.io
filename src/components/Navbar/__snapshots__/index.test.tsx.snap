// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`NavBar component > about page snapshot 1`] = `
"<nav
  className="navbar"
>
  <ul
    className="navbar-list"
  >
    <NavEntry
      isActive={true}
      page="about"
    />
    <NavEntry
      isActive={false}
      page="resume"
    />
    <NavEntry
      isActive={false}
      page="portfolio"
    />
    <NavEntry
      isActive={false}
      page="demo"
    />
    <NavEntry
      isActive={false}
      page="contact"
    />
  </ul>
</nav>"
`;
